### PR TITLE
Added target name to output path to support multiple builds of same module

### PR DESCRIPTION
**Pull request by tymurmustafaiev**
This PR is a revert of https://github.com/bazelbuild/rules_swift/pull/1097 (which was a revert of https://github.com/bazelbuild/rules_swift/pull/1082) with additional fixes 🙏 
There was an issue when without enabling the `SWIFT_FEATURE_ADD_TARGET_NAME_TO_OUTPUT` feature, target_name were added to output path of swift binary.

### CODE REVIEW

File: swift/internal/swift_binary_test.bzl [hunk 1]
```diff
@@ -343,9 +349,14 @@ def _swift_binary_impl(ctx):
...
 
+    add_target_name_to_output_path = swift_common.is_enabled(
+        feature_configuration = feature_configuration,
+        feature_name = SWIFT_FEATURE_ADD_TARGET_NAME_TO_OUTPUT,
+    )
+
     _, linking_outputs, providers = _swift_linking_rule_impl(
         ctx,
-        binary_path = ctx.label.name,
+        binary_path = derived_files.path(ctx, add_target_name_to_output_path, ctx.label.name),
```
Comment (tymurmustafaiev) [inline swift/internal/swift_binary_test.bzl:359]: @keith here was the issue. In my PR I forgot to add feature check:
`binary_path = derived_files.path(ctx, ctx.label.name),`

Comment (keith) [reply swift/internal/swift_binary_test.bzl:359]: thanks, can you add a test for this case?

Comment (tymurmustafaiev) [reply swift/internal/swift_binary_test.bzl]: @keith hey, finally [got there](https://github.com/bazelbuild/rules_swift/pull/1098/files#diff-473e214986eaa53debc16aab6d7d8eb75d9fca4e99b99d6a7b62b4784d97eb6aR1)
Added test for this case as you asked 🙏

Comment (tymurmustafaiev) [reply swift/internal/swift_binary_test.bzl]: @brentleyjones @keith 👋 🙏

**Review by keith — APPROVED** [review]

**Pull request merged**